ezgui/src/backend_glow.rs

```diff
@@ -164,16 +165,87 @@ impl<'a> GfxCtxInnards<'a> {
 }
 
 // Something that's been sent to the GPU already.
-// TODO Implement Drop; have to keep a reference to gl.
 pub struct Drawable {
-    _vert_buffer: u32,
-    vert_array: u32,
-    _elem_buffer: u32,
+    vert_array: VertexArray,
+    vert_buffer: Buffer,
+    elem_buffer: Buffer,
     num_indices: i32,
+    gl: Rc<glow::Context>,
+}
+
+impl Drop for Drawable {
+    #[inline]
+    fn drop(&mut self) {
+        self.elem_buffer.destroy(&self.gl);
+        self.vert_buffer.destroy(&self.gl);
+        self.vert_array.destroy(&self.gl);
+    }
+}
+
+struct VertexArray {
+    id: u32,
+    was_destroyed: bool,
+}
+
+impl VertexArray {
+    fn new(gl: &glow::Context) -> VertexArray {
+        let id = unsafe { gl.create_vertex_array().unwrap() };
+        VertexArray {
+            id,
+            was_destroyed: false,
+        }
+    }
+
+    fn destroy(&mut self, gl: &glow::Context) {
+        assert!(!self.was_destroyed, "already destroyed");
+        self.was_destroyed = true;
+        unsafe {
+            gl.delete_vertex_array(self.id);
+        }
+    }
+}
+
+impl Drop for VertexArray {
+    fn drop(&mut self) {
+        assert!(
+            self.was_destroyed,
```

Comment:
I did this RIAA thing for the main OpenGL resources we're cycling through. It's a little heavier than strictly necessary, but I like being able to have this check here.

If it's controversial we could just do the simplest and inline the gl.destory calls in `Drawable#drop` and get rid of these structs.